packages/wdio-cucumber-framework/src/cucumberEventListener.ts

```diff
@@ -253,15 +257,26 @@ export default class CucumberEventListener extends EventEmitter {
 
         const doc = this._gherkinDocEvents.find(gde => gde.uri === scenario?.uri)
 
+        if (this._currentDoc !== {}) {
+            if (this._currentDoc?.uri && this._currentDoc?.uri !== doc?.uri) {
```

Comment:
I tried un-nested `if` statements here but TypeScript was complaining about lots of things.